docs/docs/applying.md

```diff
@@ -0,0 +1,117 @@
+you have to think about whether it makes sense for your users to have a
+`Foo<@Nullable String>`. If that's reasonable, then annotate that type
+parameter's bound with `@Nullable`. If the type parameter has no apparent bound
+(as in `class Foo<T>`), then remember that `<T>` really means `<T extends
+Object>`, and change it to `class Foo<T extends @Nullable Object>`. If not,
+leave the bound non-nullable, and then users won't be able to declare
+`Foo<@Nullable String>` without their nullness checker reporting it to them.
+
+Then look for individual usages of each type variable that must be nullable even
+when the type argument isn't. Maybe `bar` should accept `null` even for
+`Foo<@NonNull String>`. If so, add `@Nullable` to that type variable usage:
+`bar(@Nullable T t)`.
+
+If the type parameter's bound is nullable, but you find a specific usage of that
+type variable should *not* be nullable, then add `@NonNull` to the type variable
+usage itself. For example, you know that `Optional`'s type parameter can never
+be nullable, so you would declare `bar` to return `Optional<@NonNull T>` to
+indicate that it returns `Optional<String>` even for `Foo<@Nullable String>`.
+
+(There are other edge cases, such as wildcards. Consult the
+[User Guide](/docs/user-guide) for more information.)
+
+## 2. Add `@NullMarked`
+
+Add `@NullMarked` to the class or package you're annotating to indicate that the
+remaining unannotated type usages are not nullable.
+
+If there are parts of the class or package that you're not ready to annotate
+yet, you can use `@NullUnmarked` to leave their unannotated type usages
+unspecified.
+
+## 3. Run nullness analysis on the annotated code
+
+Then run a JSpecify-aware nullness analyzer on the code you just annotated. If
+it finds any nullness problems, resolve them and rerun:
+
+*   Maybe you missed an annotation: something should be `@Nullable` or
+    `@NonNull` that isn't?
+*   Maybe your implementation code needs some annotations (type arguments in
+    local variable declarations or cast expressions)?
+*   Maybe you have a case where static analysis won't be able to tell that
+    something is null-safe, but you know it is. For example, if you call a
+    method that you know won't return `null` *for these arguments*, then you
+    might just dereference the returned value—but your analyzer might still
+    complain.
+    *   In this case, it makes sense to suppress the nullness error from your
+        analyzer with a `@SuppressWarnings` annotation and document why your
+        logic is safe anyway.
+*   Maybe your code has an actual nullness bug? If so, annotating your code
+    helped you find it!
+    *   If you can't fix the bug right now, your nullness analysis tool should
+        allow you to suppress the error with a `@SuppressWarnings` annotation,
+        or some other mechanism that lets you keep track of these as TODOs.
+*   Maybe your nullness analyzer has a bug? (This is unlikely, but possible
```

Comment:
I am trying to decide whether "unlikely" is too strong a word at this point... :)(

Reply:
How's this?